rust/kernel/error.rs

```diff
@@ -61,6 +64,26 @@ impl Error {
     }
 }
 
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // SAFETY: FFI call.
+        #[cfg(CONFIG_SYMBOLIC_ERRNAME)]
+        let name = unsafe { crate::bindings::errname(-self.0) };
+        #[cfg(not(CONFIG_SYMBOLIC_ERRNAME))]
+        let name: *const c_types::c_char = core::ptr::null();
+        if !name.is_null() {
+            // SAFETY: 'static string from C, and is not NULL.
+            let cstr = unsafe { CStr::from_char_ptr(name) };
+            // SAFETY: These strings are ASCII-only
+            let str = unsafe { str::from_utf8_unchecked(&cstr) };
+            f.debug_tuple(str).finish()
+        } else {
+            // Print out number if no name can be found.
+            f.debug_tuple("Error").field(&-self.0).finish()
+        }
```

Comment:
Invert condition and make this a `return`? That way we can remove an indentation for the happy path.